### ISSUE TITLE
Add support for handling config structs' dependency (tag "yaml inline")

### ISSUE THREAD

**Issue by luozhaoyu** [uber-go/config#59]
It would duplicate a lot if several config structs are sharing the same set of fields, so that support for struct embedding would be very useful.

One solution is to add support for yaml tag "inline", which will make these codes works:

```go
package main

import (
	"fmt"
	"log"

	"go.uber.org/config"
)

type Nested struct {
	N string
}

func main() {
	p, _ := config.NewStaticProvider(map[string]interface{}{
		"example": map[string]interface{}{
			"name":    "uber",
			"founded": 2009,
			"n":       "nnnn",
		},
	})

	var c struct {
		Nested  `yaml:",inline"`
		Name    string
		Founded int
	}

	if err := p.Get("example").Populate(&c); err != nil {
		log.Fatal(err)
	}

	fmt.Println(c)
}
```

**Comment by glibsm** [comment]
Thanks for the suggestion @luozhaoyu, we're going to look into it.

I took the liberty to edit the code in issue to make sure it runs and compiles with the latest version of the config library (import path `go.uber.org/config`).

**Comment by hsuber** [comment]
In the mean time, can we please make the config loader fail fast if it encounters an 'inline' flag?  At least it wasn't clear to me and I spent a chunk of time debugging.

**Comment by akshayjshah** [comment]
Fixed in v1.2.0.
